src/libs/ToDo.js

```diff
@@ -1,40 +1,62 @@
+/* eslint-disable import/no-cycle */
 import Task from './Task.js';
 import Storage from './Storage.js';
 import Renderer from './Renderer.js';
 
-const dummyArray = [
-  {
-    description: 'wash the dishes',
-    completed: false,
-    index: 1,
-  },
-  {
-    description: 'complete To Do list project',
-    completed: false,
-    index: 2,
-  },
-];
-
 export default class ToDo {
   constructor() {
     this.storage = new Storage('todo');
-    this.index = new Storage('indexTrack');
-    this.list = this.storage.get() || dummyArray;
-    this.storage.set(this.list);
+    this.list = this.storage.get() || [];
+    this.index = this.list.length > 0 ? this.list.length : 0;
     this.render = new Renderer('#list');
+    this.syncUpdates();
+  }
+
+  serialize() {
+    this.list.forEach((it, index) => {
+      it.index = index + 1;
+    });
+  }
+
+  syncUpdates() {
+    this.serialize();
+    this.storage.set(this.list);
+    this.render.render();
+  }
+
+  toggleTask(index) {
+    this.list.forEach((task) => {
+      if (task.index === Number(index)) {
+        task.completed = !task.completed;
+      }
+    });
+    this.syncUpdates();
   }
 
   add(task) {
-    const indexNumber = Number(this.index.get() + 1);
-    this.list.push(new Task(task.description, indexNumber));
-    this.index.set(Number(this.index.get() + 1));
-    this.render.render(this.list);
+    const indexNumber = Number(this.index);
+    this.list.push(new Task(task, indexNumber));
+    this.syncUpdates();
+  }
+
+  update(id, desc) {
+    const index = this.list.findIndex((task) => task.index === Number(id));
+    this.list[index] = { ...this.list[index], description: desc };
+    this.syncUpdates();
   }
```

Comment:
Nice job storing all the add and remove changes in local storage👍. However, according to the project requirements, all your app changes should be saved in local storage. This also includes editing a task. Currently this functionality is not implemented. Please add it🙏.

Reply:
did you read the 'syncUpdates()' method?

Reply:
@mahabubx7, kindly double-check your work. When I edit a task, the info is not saved in local storage, as seen in the video below:

https://user-images.githubusercontent.com/37116322/228758297-1c8d43aa-2385-4db6-a1ad-9dfe45fd2871.mp4

Reply:
@mahabubx7, the reviewer who approved your project did not check this. I highly encourage you to look into this and fix it so your project will be perfect